src/mscorlib/src/System/Text/Encoding.cs

```diff
@@ -722,10 +722,31 @@ public virtual String EncodingName
         {
             get
             {
-                return (Environment.GetResourceString("Globalization.cp_" + m_codePage));
+                return Environment.GetResourceString(GetEncodingNameResourceKey(m_codePage));
             }
         }
 
+        private static string GetEncodingNameResourceKey(int codePage)
+        {
+            const string KeyPrefix = "Globalization.cp_";
+
+            // Avoid intermediate `codePage.ToString()` allocations for common
+            // code pages by returning pre-concatenated string constants.
+            switch (codePage)
+            {
+                case 1200: return KeyPrefix + "1200";
+                case 1201: return KeyPrefix + "1201";
+                case 12000: return KeyPrefix + "12000";
+                case 12001: return KeyPrefix + "12001";
+                case 20127: return KeyPrefix + "20127";
+                case 28591: return KeyPrefix + "28591";
+                case 65000: return KeyPrefix + "65000";
+                case 65001: return KeyPrefix + "65001";
+            }
```

Comment:
Note: The code pages in the switch are based on:
 - [Resources included for CoreCLR](https://github.com/dotnet/coreclr/blob/b799cf9626020a0b67676e34cc9e6aeac798a87e/src/mscorlib/src/System.Private.CoreLib.txt#L3337-L3344)
 - [CoreRT implementation](https://github.com/dotnet/corert/blob/9cc058e4c7c0acf71cae83ce948d280a444bcbda/src/System.Private.CoreLib/src/System/Text/Encoding.cs#L487-L501)